### PR TITLE
fix: disable EffSymCompleter when the user is not typing a qualified name

### PR DESCRIPTION
**Pull request by stormckey**
As suggested in #9434
Preview:
image

### CODE REVIEW

**Comment by magnus-madsen** [comment]
I am not sure I understand. Should it not check whether that specific effect is in scope?

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala [hunk 1]
```diff
@@ -84,8 +84,7 @@ object CompletionProvider {
       //
       // Types.
       //
-      case SyntacticContext.Type.Eff => EffSymCompleter.getCompletions()
-      case SyntacticContext.Type.OtherType => TypeCompleter.getCompletions(ctx) ++ EffSymCompleter.getCompletions()
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala:88]: Also, are we sure we are covering all cases here?

Comment (stormckey) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala:88]: Yes, the only way a `SyntacticContext.Type.*` is built is from `UndefinedType`.
https://github.com/flix/flix/blob/983766f6689ad454e6d4f1d5757d154d15f62dc9/main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala#L216C1-L216C93
So we just move the EffSymCompleter to SemanticCompleter. As a result, EffSymCompleter can access the `err.qn` and tell if the undefined type is qualified.

**Comment by stormckey** [comment]
> I am not sure I understand. Should it not check whether that specific effect is in scope?

You're right. Actually we will disable this completer when the user is typing a unqualified name. At that time, it's AutoUseCompleter and LocalScopeCompleter to do their work.

**Comment by stormckey** [comment]
image
image
image

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**